src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> currentUser = userService.findByEmail(email);
+        if (currentUser.isEmpty() || !currentUser.get().getPassword()
+                .equals(HashUtil.hashPassword(password, currentUser.get().getSalt()))) {
+            throw new AuthenticationException("User with email: " + email
```

Comment:
better have more abstract message here, for example: "Email or password is incorrect"